aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/CredentialStoreClient.kt

```diff
@@ -56,9 +56,9 @@ internal class CredentialStoreClient(configuration: AuthConfiguration, context:
     ) {
         var capturedSuccess: Result<AmplifyCredential>? = null
         var capturedError: Exception? = null
-        val token = StateChangeListenerToken()
+        var token: StateChangeListenerToken? = StateChangeListenerToken()
         credentialStoreStateMachine.listen(
-            token,
+            token as StateChangeListenerToken,
```

Comment:
Maybe wrap the token call in a `?.let` so you don't have to do a force cast.

Reply:
Or a simple `!!`

Reply:
Done